File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int getMinimumDifference(TreeNode root) {
        List<Integer>l=new ArrayList<>();
        if(root==null){
            return 0;
        }
       
        inorder(root,l);
          int minDiff = Integer.MAX_VALUE;
        for (int i = 1; i < l.size(); i++) {
            minDiff = Math.min(minDiff, l.get(i) - l.get(i - 1));
        }
        return minDiff;
        
    }
    public void inorder(TreeNode root,List<Integer>l){
        if(root==null){
            return ;
        }
        if(root.left!=null){
            inorder(root.left,l);
        }
        l.add(root.val);
        if(root.right!=null){
            inorder(root.right,l);
        }
    }
}

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        List=[]
        self.inorder(List,root)
        mindif=10**8
        for i in range(1,len(List)):
            mindif=min(mindif,List[i]-List[i-1])
        return mindif
    def inorder(self,List,root):
        if root is None:
            return
        self.inorder(List,root.left)
        List.append(root.val)
        self.inorder(List,root.right)
        